.azure_pipelines/job_templates/build-verify-publish-template-unix.yml

```
@@ -0,0 +1,125 @@
+# PyTorch build steps template with Unix images Azure DevOps Instances
+
+parameters:
+  name: ''
+  pool: ''
+  container_endpoint: ''
+  os: ''
+  cuda: ''
+  is_ci_build: False
+  is_official_build: False
+  is_daily_build: False
+  build_stage: False
+  verify_stage: False
+  publish_stage: False
+  customMatrixes: ''
+
+jobs:
+- job: ${{parameters.name}}
+  timeoutInMinutes: 300
+  strategy:
+    matrix:
+      ${{ insert }}: ${{parameters.customMatrixes}}
+  pool:
+    name: ${{ parameters.pool}}
+  variables:
+    DECODE_PERCENTS: false
+  container:
+    image: $[variables['container_image']]
+    endpoint: ${{parameters.container_endpoint}}
+
+  steps:
+  # Build stage
+  - ${{ if eq(parameters.build_stage, 'True') }}:
+    # Set up environment variables for specific pipeline build
+    - template: set-environment-variables.yml
+      parameters:
+        os: ${{ parameters.os}}
+        cuda: ${{ parameters.cuda}}
+        is_official_build: ${{ parameters.is_official_build}}
+
+    # Sync and update PyTorch submodules
+    - bash: git submodule update --init --recursive
+      displayName: Update PyTorch submodules
+
+    # Build PyTorch and run unit tests - no packaging
+    - ${{ if or(eq(parameters.is_ci_build, 'True'), eq(parameters.is_daily_build, 'True')) }}:
+      # Build PyTorch from source in develop mode
+      - bash: python setup.py develop
+        displayName: Build PyTorch from source
+
+      - ${{ if eq(parameters.is_ci_build, 'True') }}:
+        # Run TestTorch unit tests to demonstrate successful PyTorch build
+        - bash: python test/test_torch.py TestTorch
+          displayName: Run TestTorch unit tests
+
+      - ${{ if eq(parameters.is_daily_build, 'True') }}:
+        # Run all unit tests to demonstrate successful PyTorch build
+        - bash: python test/run_test.py --continue-through-error --exclude-jit-executor --shard 1 2 --verbose
```

Please remove `--shard 1 2` option if you really want to run all unit tests, otherwise you are running only half of them (i.e. shard 1 out of 2)
```suggestion
        - bash: python test/run_test.py --continue-through-error --exclude-jit-executor --verbose
```